instances/expander.go

```diff
@@ -136,13 +136,33 @@ func (e *Expander) ExpandResource(parentAddr addrs.Module, resourceAddr addrs.Re
 	// (moduleInstances does plenty of allocations itself, so the benefit of
 	// pre-allocating this is marginal but it's not hard to do.)
 	moduleInstanceAddr := make(addrs.ModuleInstance, 0, 4)
-	ret := e.exps.resourceInstances(parentAddr, resourceAddr, moduleInstanceAddr)
+	ret := e.exps.resourceInstances(moduleAddr, resourceAddr, moduleInstanceAddr)
 	sort.SliceStable(ret, func(i, j int) bool {
 		return ret[i].Less(ret[j])
 	})
 	return ret
 }
 
+// ExpandResource finds the set of resource instances resulting from
+// the expansion of the given resource within its module instance.
+//
+// All of the modules on the path to the identified resource and the resource
+// itself must already have had their expansion registered using one of the
+// SetModule*/SetResource* methods before calling, or this method will panic.
+func (e *Expander) ExpandResource(resourceAddr addrs.AbsResource) []addrs.AbsResourceInstance {
```

Comment:
Should this have a lock involved, and if not, why not?

Reply:
Locking here would deadlock, as this is delegating to the `ExpandModuleResource` which is already locked.

Reply:
Ahhh yes. Thanks!